contents/testing/integration-testing.md

```diff
@@ -0,0 +1,143 @@
+
+## Tips for better integration tests
+
+### :thumbsup: Make sure that each component is unit tested before integration testing
+By ensuring that each unit test is completed properly, integration testing will be smoother as we can focus mainly on the flow of data between modules.
+
+### :thumbsup: Prioritize the modules to be tested
+Despite the need to cover all areas of integration of the application, it is important to ensure that critical modules needs to be tested first.
+
+### :thumbsup: Automate your tests (strongly encouraged)
+As far as possible, automate all your tests, especially when you use the incremental approach, since regression testing is important each time you integrate a unit, and manual regression testing can be inefficient. You can find a list of most commonly used [automation tools](https://medium.com/@briananderson2209/best-automation-testing-tools-for-2018-top-10-reviews-8a4a19f664d2) for integration testing.
+
+### :thumbsup: Ensure that all executed test cases are documented
+This helps you to identify errors quickly through an integration test. It also helps to standadize the way integration testing is carried out in your application so that everyone can conform to the given standard.
+
+## Tools to get started with integration testing
+
+Here are some useful tools that you can use for your integration testing:
+
+- [Selenium](https://www.seleniumhq.org/). Selenium is an open source test automation framework focusing on web applications. It supports a wide range of programming languages, cross-browser testing with extensive libraries and the ability to create robust test scripts to handle many scenarios.
+
+- [Google EarlGrey](https://google.github.io/EarlGrey/). EarlGrey is a native iOS automation test framework allowing developers to write and maintain clear concise tests. It has a powerful built-in synchornization which allows it to reproduce any UI interactivity and test them.
+
+- [Robotium](https://github.com/RobotiumTech/robotium). Robotium is an Android test automation framework made to simplify black-box tests/ It can handle multiple activities atuomatically, producing fast and robust tests cases.
+
+- [Gauge](https://gauge.org/). Gauge acts as a plugin which can be incorporated to any language or IDE. It is an lightweight cross-platform test automation tool which makes testing easier to maintain, more readable and flexible.
+
+Note that although there are many integration testing tool available, more research needs to be conducted to ensure the compatibility of the tool with your application.
+
+## Concluding Remarks
+
+Ultimately, as a developer, it is important to recognize the importance of integration testing in your application. To better complete integration test, it is important to put yourself in the users' perspective and ensure that your application works from the user's point of view. 
```

Comment:
> complete integration test

Fix grammar

> it is important to put yourself in the users' perspective 

Why? This seems to be the first time that this point is brought up in this chapter